### ISSUE TITLE
Can't create clusterroletemplatebindings with AD user

### ISSUE THREAD

**Issue by azbpa** [rancher/rancher#18990]
**Bug**

From API explorer I am able to create a new `clusterRoleTemplateBinding`:

*cURL command line:*
```
curl -u "${CATTLE_ACCESS_KEY}:${CATTLE_SECRET_KEY}" \
-X POST \
-H 'Accept: application/json' \
-H 'Content-Type: application/json' \
-d '{"clusterId":"c-vfcf5", "name":"", "namespaceId":"", "roleTemplateId":"cluster-admin", "userPrincipalId":"activedirectory_user://CN=myuser, OU=user, DC=domian,
 DC=local"}' \
'https://rancher.domain.local/v3/clusterroletemplatebindings'
```

*HTTP Request:*
```
HTTP/1.1 POST /v3/clusterroletemplatebindings
Host: rancher.domain.local
Accept: application/json
Content-Type: application/json
Content-Length: 174

{
"clusterId": "c-vfcf5",
"name": "",
"namespaceId": "",
"roleTemplateId": "cluster-admin",
"userPrincipalId": "activedirectory_user://CN=myuser,OU=user,DC=domain,DC=local"
}
```
*HTTP Response:*

```
HTTP/1.1 201
date: Mon, 18 Mar 2019 15:29:47 GMT
x-api-schemas: https://rancher.domain.local/v3/schemas
content-type: application/json
content-length: 836
expires: Wed 24 Feb 1982 18:42:00 GMT

{
"annotations": {
"auth.cattle.io/principal-display-name": "myuser"
},
"baseType": "clusterRoleTemplateBinding",
"clusterId": "c-vfcf5",
"created": "2019-03-18T15:29:47Z",
"createdTS": 1552922987000,
"creatorId": "user-dkjn4",
"groupId": null,
"groupPrincipalId": null,
"id": "c-vfcf5:crtb-ww69k",
"labels": {
"cattle.io/creator": "norman"
},
"links": {
"remove": "…/v3/clusterRoleTemplateBindings/c-vfcf5:crtb-ww69k",
"self": "…/v3/clusterRoleTemplateBindings/c-vfcf5:crtb-ww69k",
"update": "…/v3/clusterRoleTemplateBindings/c-vfcf5:crtb-ww69k"
},
"name": "crtb-ww69k",
"namespaceId": null,
"roleTemplateId": "cluster-admin",
"type": "clusterRoleTemplateBinding",
"userId": null,
"userPrincipalId": "activedirectory_user://CN=myuser,OU=user,DC=domain,DC=local",
"uuid": "a9dda9e1-4992-11e9-85c8-0242ac110002"
}
```

However, when I try to reproduce the same with `curl` (or `PowerShell` or any other REST client) using the same access/secret keys and the command generated above (*cURL command line*), I am getting the following error:

```
{  
"baseType": "error",  
"code": "ServerError",  
"message": "user.management.cattle.io \"CN=myuser, OU=user,  DC=domain, DC=local\" not found",  
"status": 500,  
"type": "error"  
}
```
Also tried to remove the "space" characters from the DN but this ends in the same result.

**Environment information**
- Rancher version `rancher/rancher:v2.1.6` 
- Installation option: single
- MS AD for authentication

**Comment by cjellick** [comment]
Do you have CATTLE_ACCESSS_KEY and CATTLE_SECRET_KEY exported as environment variables? That curl command assumes that you do

**Comment by azbpa** [comment]
Yes, sure. Communication is working. 

Further I noticed that i.e. `search` from the API Explorer and the curl command seems to have also a different behavior. 

From the API Explorer it is working correctly. From the curl command it is only returning `local` user accounts and none from source `active_directory`.

So somehow it seems to be an context issue.

I noticed that my domain account is somehow related to the local Default Admin account (shown under `Preferences` in user drop down) but is displayed as my domain account name. The domain account is also not shown in the overall `Users` list.

![image](https://user-images.githubusercontent.com/2796467/54593409-ca083f80-4a2e-11e9-96de-936806047a66.png)


When I use the local credentials for Default Admin I can verify that I am seeing the same Access Tokens. With the difference that I can't resolve `active_directory` accounts from the UI:

![image](https://user-images.githubusercontent.com/2796467/54593388-bceb5080-4a2e-11e9-9c6f-4e493b5f9a17.png)
